submissions/BlessingOkoro/ecommerce-project/src/Application/Actions/AddProductsAction.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Application\Actions;
+
+use App\Application\Handlers\DbHandlerInterface;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Log\LoggerInterface;
+
+class AddProductsAction extends Action
+{
+    /** @var \PDO $connection */
+    private $connection;
+
+    public function __construct(LoggerInterface $logger, DbHandlerInterface $dbHandlerInterface)
+    {
+        parent::__construct($logger);
+        $this->connection = $dbHandlerInterface->getConnection();
+    }
+    /**
+     * @param Request  $request
+     * @param Response $response
+     */
+    public function action(){
+
+        if (isset($_POST['value'])){
```

Comment:
You can leverage the Request in the base class by accessing POST data like this:
```
        $params = $this->request->getParsedBody();
        $itemName = $params["itemname"];
        $quantity = $params["quantity"];
```
etc.